dace/codegen/targets/rtl.py

```diff
@@ -328,6 +356,130 @@ def generate_cpp_internal_state(self, tasklet):
+            {outputs}
+            model->m_axis_{name}_tready = 1;
+        }}'''
+        return [
+            template.format(name=name,
+                            debug_export_element=debug_export_element,
+                            outputs=outputs) for name in tasklet.out_connectors
+        ]
+
+    def generate_write_output_hs(self, tasklet):
+        """
+        Generate check for whether an element has been consumed from the output of a tasklet.
+        """
+        template = '''
+        if (model->m_axis_{name}_tready && model->m_axis_{name}_tvalid == 1) {{
+            write_output_hs_{name} = true;
+        }}'''
+        return [template.format(name=name) for name in tasklet.out_connectors]
+
+    def generate_hs_flags(self, tasklet):
+        """
+        Generate flags
+        """
+        ins = [
+            f'bool read_input_hs_{name} = false;'
+            for name in tasklet.in_connectors
+        ]
+        outs = [
+            f'bool write_output_hs_{name} = false;'
+            for name in tasklet.out_connectors
+        ]
+        return ins + outs
+
+    def generate_input_hs_toggle(self, tasklet):
+        """
+        Generate statements for toggling input flags.
+        """
+        debug_read_input_hs = "std::cout << \"remove read_input_hs flag\" << std::endl;" if self.verilator_debug else ""
+        template = '''
+        if (read_input_hs_{name}) {{
+            // remove valid flag
+            model->s_axis_{name}_tvalid = 0;
+            {debug_read_input_hs}
+            read_input_hs_{name} = false;
+        }}'''
+        return [
+            template.format(name=name, debug_read_input_hs=debug_read_input_hs)
+            for name in tasklet.in_connectors
+        ]
+
+    def generate_output_hs_toggle(self, tasklet):
+        """
+        Generate statements for toggling output flags.
+        """
+        debug_write_output_hs = "std::cout << \"remove write_output_hs flag\" << std::endl;" if self.verilator_debug else ""
+        template = '''
+        if (write_output_hs_{name}) {{
+            // remove ready flag
+            model->m_axis_{name}_tready = 0;
+            {debug_write_output_hs}
+            write_output_hs_{name} = false;
+        }}'''
+        return [
+            template.format(name=name,
+                            debug_write_output_hs=debug_write_output_hs)
+            for name in tasklet.out_connectors
+        ]
+
+    def generate_running_condition(self, tasklet):
+        """
+        Generate the condition for whether the simulation should be running.
+        """
+        # TODO should be changed with free-running kernels. Currently only
+        # one element is supported. Additionally, this should not be used as
+        # condition, as the amount of input and output elements might not be
+        # equal to each other.
+        return [
+            f'out_ptr_{name} < num_elements_{name}'
+            for name in tasklet.out_connectors
+        ]
```

Comment:
Is this still true? Is this a Verilator thing?

Reply:
It's a Verilator thing and will be changed after implementing free running kernels in Verilator.